MAINTAINERS.md

```diff
@@ -0,0 +1,197 @@
+1. The PR is at least 1 week old, to permit time for review.
+1. The maintainer should communicate intent to self-merge 48 hours before doing so
+
+These rules aren't hard and fast, but are considered a best practice to avoid
+self-merge as a convenience.
+
+
+
+## Nomination and Selection of Maintainers
+
+Being a Maintainer is not an honor (but it may be honorable to serve as one).
+Many regular contributors who are capable or experienced enough choose not to
+be a Maintainer. Similarly, just because a regular contributor has reached a
+certain level of experience does not mean that they "deserve" to be a maintainer.
+
+Maintainers are added when there is a demonstrated need. For there to be a new
+maintainer, first an issue should be created with a "Call for Maintainer" (C4M)
+by one of the existing maintainers or regular contributors. C4Ms should specify
+if the call is for a General Maintainer, or for a Scoped Maintainer. If the
+role is for a Scoped Maintainer, the C4M should describe the need for a new
+maintainer, and link to some Pull Requests (or other material) that would be
+relevant for such a maintainer to be responsible for.
+
+Following the C4M, anyone may respond to the issue with a nomination for the
+position and a description of their qualifications. Self-nomination is
+encouraged. Nominations must be accepted. After 2 weeks from the date of issue
+creation, assuming there is at least 1 nomination, the maintainers may select
+one or more candidate from the nominee pool at their discretion. The selected
+candidate should create a PR for adding their key to the repository and to the
+above table (see https://github.com/bitcoin/bitcoin/pull/23798 for example).
+During / Before the PR, regular contributors are encouraged to correspond with
+candidate(s) publicly about any concerns or questions. Candidates are also
+encouraged to write up a Prioritization statement (see below) in conjunction
+with the key PR.
+
+At the discretion of the maintainers, and no sooner than 2 weeks after the PR
+is opened, the candidate(s) PR may be merged.
+
+
+## Maintainer Prioritizations
+
+While the maintainers are ultimately not accountable to any particular roadmap,
+nor can make regular contributors focus on one area or another, practically the
+views and prioritizations of maintainers have an impact on what gets reviewed,
+merged, and the general direction of the project.
+
+As such, maintainers should regularly deliver a written "state of the union"
+article on what they intend to focus on in the coming period. Such addresses
+should be done per-release cycle, and should minimally cover:
+
+1. What went well in the last release cycle
+1. What is making good progress for this release cycle
+1. How the maintainer intends to prioritize outstanding
+   PRs/issues/features/projects in the next release cycle
+1. Projects which could use participation from other contributors
+1. Longer term focuses beyond the next release
```

Comment:
I don't see how this could be of any help. If you want to find out what went well in the past, you can read the release notes. If you want maintainers to prioritize something, it likely won't work. (Assume I prioritized $project, but then no one reviewed it, am I supposed to merge it without review?)

Don't get me wrong, I am not holding anyone back from sharing their interests and motivations or making a personal roadmap, but in my eyes this has nothing to do with maintainers. In fact, if you force contributors that are maintainers to come up with a roadmap, I could see that this will discourage contributors that are non-maintainers from sharing their goals, since it doesn't align with the "official roadmap".

Reply:
In my experience, both from corporate projects and open source projects, including Bitcoin Core, this would be mostly bureaucracy stuff with very little or no real impact. Are maintainers doing this now? If not, then why impose that on them with this document?

Also, it gives somehow the false impression that maintainers are deciding there the project is going. In reality contributors do by opening PRs and reviewing them.

Reply:
+1 to the other commenters, the only thing id add:

> practically the views and prioritizations of maintainers have an impact on what gets reviewed, merged, and the general direction of the project.

this seems like a bug-not-a-feature. if anything, we should be thinking of ways to minimize this, not formalize it.

Reply:
> I don't see how this could be of any help. If you want to find out what went well in the past, you can read the release notes. If you want maintainers to prioritize something, it likely won't work. (Assume I prioritized $project, but then no one reviewed it, am I supposed to merge it without review?)

Reading release notes is not really a valid summary of what went well, since release notes don't contain any information on e.g., what went according to expectations, what complexities came up, status of ongoing projects.

> Don't get me wrong, I am not holding anyone back from sharing their interests and motivations or making a personal roadmap, but in my eyes this has nothing to do with maintainers. In fact, if you force contributors that are maintainers to come up with a roadmap, I could see that this will discourage contributors that are non-maintainers from sharing their goals, since it doesn't align with the "official roadmap".

There is a section in the document encouraging regular contributors to share their prioritization documents listed alongside the maintainers.

Reply:
> In my experience, both from corporate projects and open source projects, including Bitcoin Core, this would be mostly bureaucracy stuff with very little or no real impact. Are maintainers doing this now? If not, then why impose that on them with this document?
> 
> Also, it gives somehow the false impression that maintainers are deciding there the project is going. In reality contributors do by opening PRs and reviewing them.

It's something that's been done historically by past maintainers, but I think it's never been codified as a requirement. 

The point of imposing it as such is that maintainers already _do_ have these prioritizations and plans, but they live in their heads and aren't shared publicly with opportunity for comment.

Having maintainers regularly write up their thoughts on what they think is possible / prioritized and how they will spend their time leads to better utilization of all contributors time, as well as helps contributors understand if the projects they are working on are unlikely to recieve 'support' from maintainers (which might open the door to proposing a new maintainer with that scope).

Reply:
> if the projects they are working on are unlikely to recieve 'support' from maintainers (which might open the door to proposing a new maintainer with that scope).

If your project doesn't receive any review, then an encouragement by maintainers is likely not going to magically give it reviewers. If a project isn't merged despite review, then it may have outstanding issues or controversies. Assigning a new maintainer for the sole purpose of merging your project isn't magically going to resolve them.

Reply:
do you mind de-personalizing this? i don't presently have any work i think would be benefitted by the addition of a maintainer.

There are many contributors who are not maintainers who have long term projects that seem to be somewhat stalled out, even though there are other individual who might be qualified to serve as a scoped maintainer to help (or that individual could as well). I think often times the issue in it of itself is getting sufficient review, and maintainers with scopes help coordinate review. And there are PRs where there is sufficient review, but no maintainer has a strong interest that stall out. These are circumstances where adding a maintainer would help.

Reply:
> These are circumstances where adding a maintainer would help.

Yes, I agree that currently adding more maintainers is probably beneficial to the project. Especially if making someone maintainer "tricks" them into spending time on code review in the areas they are interested in. Moreover, when there are multiple maintainers for the same module, it might naturally increase the accountability and cross-checks, something that you'd rather see fixed by having them ["check some boxes"](https://github.com/bitcoin/bitcoin/pull/25560#discussion_r916186988).